app/models/sign_up_team.rb

```diff
@@ -0,0 +1,46 @@
+class SignUpTeam < ApplicationRecord
+  belongs_to :sign_up_topic, counter_cache:true
+  belongs_to :team
+  
+  validates :is_waitlisted, presence: true 
+
+  # Get participants in the team signed up for the given topic.
+  def find_team_participants()
+    return team.get_team_participants()
+  end
+
+  def release_topics()
+    # TODO: Update waitlist and release topics.
+  end
+
+  #Create a signUp team with the given parameters topic_id and is_waitlisted 
+  def self.create_sign_up_team(is_waitlisted, topic_identifier, teams_id)
+    sign_up_topic = SignUpTopic.where(topic_identifier: topic_identifier).first
+
+    if sign_up_topic.find_if_topic_available? == false
+      return
```

Comment:
does it need to return a error so that the controller calling this method can report an error to the user.